network/benchmarks/netperf/nptest/tests.go

```diff
@@ -0,0 +1,332 @@
+package main
+
+import (
+	"k8s.io/perf-tests/network/nptest/parsers"
+)
+
+type TestType int
+
+const (
+	iperfTCPTest TestType = iota
+	qperfTCPTest
+	iperfUDPTest
+	iperfSctpTest
+	netperfTest
+	iperfThroughputTest
+	iperfThroughputUDPTest
+)
+
+type testcase struct {
+	TestParams
+	Label           string
+	Finished        bool
+	BandwidthParser func(string) string
+	JsonParser      func(string) string
+	// Deprecated: We will use declarative approach to define test cases
+	Type TestType
+}
+
+type TestParams struct {
+	SourceNode      string
+	DestinationNode string
+	ClusterIP       bool
+	MSS             int
+	MsgSize         int
+}
+
+var testcases = []*testcase{
+	// {
+	// 	Label: "1 qperf TCP. Same VM using Pod IP",
+	// 	TestParams: TestParams{
+	// 		SourceNode:      "netperf-w1",
+	// 		DestinationNode: "netperf-w2",
+	// 		ClusterIP:       false,
+	// 		MSS:             mssMin,
+	// 	},
+	// 	BandwidthParser: parsers.ParseIperfTCPBandwidth,
+	// 	Type:            qperfTCPTest,
+	// },
+	// {
+	// 	Label: "2 qperf TCP. Same VM using Virtual IP",
+	// 	TestParams: TestParams{
+	// 		SourceNode:      "netperf-w1",
+	// 		DestinationNode: "netperf-w2",
+	// 		ClusterIP:       true,
+	// 		MSS:             mssMin,
+	// 	},
+	// 	BandwidthParser: parsers.ParseIperfTCPBandwidth,
+	// 	Type:            qperfTCPTest,
+	// },
+	// {
+	// 	Label: "3 qperf TCP. Remote VM using Pod IP",
+	// 	TestParams: TestParams{
+	// 		SourceNode:      "netperf-w1",
+	// 		DestinationNode: "netperf-w3",
+	// 		ClusterIP:       false,
+	// 		MSS:             mssMin,
+	// 	},
+	// 	BandwidthParser: parsers.ParseIperfTCPBandwidth,
+	// 	Type:            qperfTCPTest,
+	// },
+	// {
+	// 	Label: "4 qperf TCP. Remote VM using Virtual IP",
+	// 	TestParams: TestParams{
+	// 		SourceNode:      "netperf-w3",
+	// 		DestinationNode: "netperf-w2",
+	// 		ClusterIP:       true,
+	// 		MSS:             mssMin,
+	// 	},
+	// 	BandwidthParser: parsers.ParseIperfTCPBandwidth,
+	// 	Type:            qperfTCPTest,
+	// },
+	// {
+	// 	Label: "5 qperf TCP. Hairpin Pod to own Virtual IP",
+	// 	TestParams: TestParams{
+	// 		SourceNode:      "netperf-w2",
+	// 		DestinationNode: "netperf-w2",
+	// 		ClusterIP:       true,
+	// 		MSS:             mssMin,
+	// 	},
+	// 	BandwidthParser: parsers.ParseIperfTCPBandwidth,
+	// 	Type:            qperfTCPTest,
+	// },
+
+	{
+		Label: "6 iperf TCP. Same VM using Pod IP",
```

Comment:
i am not familiar much with go testing for context

q: it appears to me as though we're just defining the input parameters for the tests - how do we know that the output is what it should be?

Reply:
It's a benchmark test, iperf/netperf are tools that gives us network performances, we would be running these tests with and without retina installed and compare the outputs.

Reply:
right, so the comparison is manual

Reply:
Ideally, we would do it from retina, these changes are on k8s org's repo so is independent of our usecase.